src/app/share-report/share-report.component.ts

```diff
@@ -221,18 +213,29 @@ export class ShareReportComponent implements AfterViewInit {
         throw new Error('Missing author screenname for comment: ' + comment);
       }
     }
-    return comments.map(
+    const users = comments.map(
       (comment): TwitterUser => ({
         id_str: comment.item.authorId!,
         screen_name: comment.item.authorScreenName!,
       })
     );
+    // Remove duplicate user IDs.
+    const userIds = new Set();
+    return users.filter((user) => {
+      const hasId = userIds.has(user.id_str);
+      userIds.add(user.id_str);
+      return !hasId;
```

Comment:
Should we somehow surface to the user that the user id is missing for these users (and therefore that user won't be blocked)? Would it be better to let the server handle these errors, since we already have error handling built in there?

Reply:
Good question. The primary motivation for filtering and throwing errors here is that the `TwitterUser` interface expects `id_str` and `screen_name` to be defined, whereas `authorId` and `authorScreenName` in the `SocialMediaItem` interface are optional. That being said, the point about letting the server handle it is a good one, so I'll play around with that and send any changes in a future PR to keep this one focused.